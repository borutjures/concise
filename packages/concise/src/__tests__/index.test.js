/* eslint-env jest */

import Concise from '..';

const schema1 = require('./schema1.json');
const schema2 = require('./schema2.json');

describe('main API', () => {
  it('should correctly merge two schemas', () => {
    const concise = new Concise();
    concise.addSchema(schema1);
    concise.addSchema(schema2);
    expect(concise.getSchema()).toMatchSnapshot();
  });
});
