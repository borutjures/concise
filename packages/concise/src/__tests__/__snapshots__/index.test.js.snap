// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main API should correctly merge two schemas 1`] = `
Object {
  "models": Object {
    "common": Object {
      "fields": Object {
        "id": Object {
          "type": "uuid",
        },
      },
    },
    "project": Object {
      "fields": Object {
        "endDate": Object {
          "time": false,
          "type": "date",
        },
        "name": Object {
          "type": "string",
        },
        "startDate": Object {
          "time": false,
          "type": "date",
        },
      },
      "includes": Object {
        "common": true,
      },
      "relations": Object {
        "creator": Object {
          "inverse": Object {
            "name": "createdProjects",
          },
          "model": "user",
        },
        "projectManager": Object {
          "model": "user",
        },
        "technicalManager": Object {
          "model": "user",
        },
      },
    },
    "user": Object {
      "fields": Object {
        "name": Object {
          "type": "string",
        },
        "surname": Object {
          "type": "string",
        },
      },
      "includes": Object {
        "common": true,
      },
    },
  },
}
`;
